### PR TITLE
Soft deprecate `file_path` for `SpikeGLXRecordingInterface` and `SpikeGLXNIDQInterface`

### LINKED ISSUE #978: Match signature of `SpikeGLXRecordingInterface` interface to `SpikeGLXRecordingExtractor` extractor
`SpikeGLXRecordingInterface` takes a file_path as an input and uses it to determine the stream whereas `SpikeGLXRecordingExtractor` takes a folder as input and uses the stream to determine the file.

I think we are kind of duplicating efforts here and it feels to me that we should strieve to keep the interafce between the two libraries consistent. Any downsides of moving in this direction that I am not seeing?

### PR DESCRIPTION
**Pull request by h-mayorquin**
As this is an important change, we are giving it  a year instead of our typical six months. 

Should close #978

### CODE REVIEW

File: src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def __init__(
         self,
         file_path: Optional[FilePath] = None,
         verbose: bool = True,
-        load_sync_channel: bool = False,
+        load_sync_channel: Optional[bool] = False,
```
Comment (bendichter) [inline src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py:39]:
```suggestion
        load_sync_channel: Optional[bool] = None,
```

Comment (bendichter) [reply src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py:39]: this is supposed to be `None`, right?

**Comment by bendichter** [comment]
lgtm

File: src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py (hunk 1 above)
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py:39]: Yes.

**Review by bendichter — APPROVED** [review]

**Pull request merged**